src/query/plan/preprocess.hpp

```diff
@@ -302,6 +301,8 @@ class Filters final {
   auto erase(iterator first, iterator last) { return all_filters_.erase(first, last); }
   auto erase(const_iterator first, const_iterator last) { return all_filters_.erase(first, last); }
 
+  void SetFilters(std::vector<FilterInfo> all_filters) { all_filters_ = all_filters; }
```

Comment:
Why not pass `all_filters` by `const &`, or move them here?

Reply:
I think this can accept rvalue, and then you move to all_filters_:
```
void SetFilters(std::vector<FilterInfo> &&all_filters) { all_filters_ = std::move(all_filters); }
```

Reply:
Added